ci/builddeps.sh

```diff
@@ -0,0 +1,110 @@
+#!/bin/bash
+# Copyright 2021 Simon McVittie
+# SPDX-License-Identifier: LGPL-2.0-or-later
+
+set -eux
+set -o pipefail
+
+usage () {
+    if [ "${1-2}" -ne 0 ]; then
+        exec >&2
+    fi
+    cat <<EOF
+Usage: see source code
+EOF
+    exit "${1-2}"
+}
+
+opt_clang=
+
+getopt_temp="help"
+getopt_temp="$getopt_temp,clang"
+
+getopt_temp="$(getopt -o '' --long "${getopt_temp}" -n "$0" -- "$@")"
+eval set -- "$getopt_temp"
+unset getopt_temp
+
+while true; do
+    case "$1" in
+        (--clang)
+            clang=yes
+            shift
+            ;;
+
+        (--help)
+            usage 0
+            # not reached
+            ;;
+
+        (--)
+            shift
+            break
+            ;;
+
+        (*)
+            echo 'Error parsing options' >&2
+            usage 2
+            ;;
+    esac
+done
+
+# No more arguments please
+for arg in "$@"; do
+    usage 2
+done
+
+if dpkg-vendor --derives-from Debian; then
```

Comment:
Not a blocker but personally I like using `yum builddep`/`apt builddep` style flows for software that's already packged to avoid the build dependency duplication.

Reply:
The problem with `apt-get build-dep` (and presumably `yum builddep` too) is that it gets the dependencies for the version that was packaged, in the configuration that was packaged. The version we're building is newer, and the configuration we're building here isn't necessarily identical to Ubuntu's.

We're running CI here on the latest Ubuntu LTS, currently 20.04 (released April 2020), until it gets superseded by 22.04 - which will be April 2022 at the earliest, plus any delay while Github make new VM images.

A more minor issue is that in addition to what's needed to compile the upstream software, `apt-get build-dep` will fetch things like `debhelper` which are needed for the packaging but not for the upstream code. This is probably not such a big deal for bubblewrap, but if we want fast CI then we wouldn't want to accidentally pull in something big like LaTeX.